Crowdsignal: updates to get ready for Fusion

jeherve opened the pull request:
#### Changes proposed in this Pull Request:

This PR aims to fix all PHPCS warnings in the Crowdsignal / Polldaddy shortcode, so those files can be synced with WordPress.com and replace the existing shortcode there.

I made a few different changes:

- Fix PHPCS warnings
- Remove `GLOBAL_TAGS` check (we don't use that constant anywhere on WordPress.com or in Jetpack anymore).
- Remove `PADPRESS_LOADED` constant check (it was used by the Onswipe plugin, but that plugin is not in the repository anymore and we don't use the constant anywhere on WordPress.com anymore either).
- Modify the way JavaScript is enqueued. Rely on core functions (we now use `wp_register_script`, `wp_enqueue_script`, and `wp_add_inline_script` to enqueue JavaScript content on the pages).

#### Testing instructions:

**To test, you'll need to either run `yarn build` or add `define( 'SCRIPT_DEBUG', true );` to your `wp-config.php` file**

* You'll want to insert different shortcodes in your posts; some posts with only one shortcode, others with multiple shortcodes at once:
 * `[polldaddy type="iframe" survey="EB151947E5950FCF" height="auto" domain="jeherve" id="a-survey-with-branches"]`
 * `[crowdsignal type="iframe" survey="EB151947E5950FCF" height="auto" domain="jeherve" id="a-survey-with-branches"]`
 * `https://polldaddy.com/poll/7910844/`
 * `https://jeherve.survey.fm/a-survey`
 * `https://jeherve.survey.fm/a-survey-with-branches`
 * `[crowdsignal type="iframe" survey="7676FB1FF2B56CE9" height="auto" domain="jeherve" id="a-survey"]`
 * `[crowdsignal survey="7676FB1FF2B56CE9"]`
 * `[polldaddy survey="7676FB1FF2B56CE9"]`
 * `[crowdsignal poll=9541291]`
 * `[crowdsignal poll=9541291 type=slider]`
 * `[crowdsignal rating=8755352]`
* You should check that the shortcodes are displayed properly on the post's page, but also on the home page. You'll find that in some scenarios (looking at the home page on a site with a theme that supports Infinite Scroll), only a link is displayed.

@donnchawp I would appreciate your review on this PR. I don't know of all the potential embed combinations one may use, so I am afraid I may have missed one. Could you take a look?

#### Remaining question:

On WordPress.com we have an additional `site` parameter for the shortcode, introduced in r181467-wpcom. I am not sure if it is still necessary, and if we should keep it. @donnchawp Could you take a look? Would it be okay to use the short domains for all embeds now?

#### Proposed changelog entry for your changes:

* None

jeherve committed:
Crowdsignal: fix phpcs warnings, round 1

jeherve committed:
Remove GLOBAL_TAGS check

We don't use that constant anywhere on WordPress.com or in Jetpack anymore.

jeherve committed:
remove comment about extract, we don't use extract anymore

jeherve committed:
Remove PADPRESS_LOADED constant check

It was used by the Onswipe plugin, but that plugin is not in the repository anymore and we
don't use the constant anywhere on WordPress.com anymore either.

jeherve committed:
Crowdsignal: minor indenting and formatting adjustments.

jeherve committed:
Modify the way JavaScript is enqueued. Rely on core functions.

- We now use wp_register_script, wp_enqueue_script, and wp_add_inline_script to enqueue
JavaScript content on the pages.

kraftbj reviewed (dismissed):
Worked well for me. Approving, but leaving as `Needs Review` pending @donnchawp's review.

donnchawp commented on modules/shortcodes/crowdsignal.php:
```diff
@@ -1,610 +1,686 @@
 /**
-* Class wrapper for Crowdsignal shortcodes
-*/
+ * Crowosignal (ne PollDaddy) shortcode.
```
Tiny typo here should be "Crowdsignal".

jeherve replied:
Fixed in 6c160d12b4d945eadc77f1691cb12eec8eb37ad8

donnchawp commented on modules/shortcodes/crowdsignal.php:
```diff
@@ -1,610 +1,686 @@
+			$attributes = shortcode_atts(
+				array(
+					'survey'     => null,
+					'link_text'  => esc_html__( 'Take Our Survey', 'jetpack' ),
```
Should this use "jetpack" as it'll be on WordPress.com?

jeherve replied:
No need. WordPress.com handles the `jetpack` textdomain well.

jeherve committed:
Fix typo

jeherve commented:
@donnchawp Do you mind commenting about this part?

> Remaining question:
On WordPress.com we have an additional site parameter for the shortcode, introduced in r181467-wpcom. I am not sure if it is still necessary, and if we should keep it. @donnchawp Could you take a look? Would it be okay to use the short domains for all embeds now?

Thank you!

kraftbj commented:
Marking for author reply as a holding status for the outstanding question.

jeherve committed:
AMP: check if the compat class exists before to use it

kraftbj reviewed (dismissed):
Ideally, find out about that parameter, but that in and of itself doesn't block the PR. Approving and entrusting @jeherve to make the call to merge for 7.3 or move out of the milestone.

jeherve committed:
Support embeds pointing to legacy domain.

ports r181467-wpcom to Jetpack.

Addresses potential issues mentioned in https://github.com/Automattic/jetpack/pull/11814#issuecomment-485939263

jeherve committed:
Add a first set of basic tests for Polldaddy and Crowdsignal polls

jeherve commented:
Noting that I ported r181467-wpcom to this PR to try to make our decision about the above easier. I also started adding more tests, and will continue to do so to try to cover more of the different configurations offered by the shortcode.

jeherve committed:
Bail early when we have an empty shortcode

jeherve committed:
Add check to be able to check the full shortcode output in tests

jeherve committed:
Update inline comments to make it easier to parse

jeherve committed:
Add more unit tests to cover some of the shortcode combinations

jeherve committed:
Extend tests to check if scripts are enqueued alongside markup

jeherve committed:
Bring back changes from cac43f58ab9b03091bb7738095a73ed2d4116779

They were inadvertently removed in ae24c42ed350c3415cca738ce8d63a2a14153a57

jeherve committed:
Add comment about slider type poll

jeherve committed:
Add test about slider type poll

jeherve committed:
Add new JS file from WordPress.com.

jeherve committed:
Add comment about the cb parameter

jeherve committed:
Check for a valid array of attributes earlier

jeherve committed:
Simplify embed URLs for iFramed surveys with custom domains

This way they now support both old legacy polldaddy shortcodes
and new crowdsignal shortcodes

jeherve committed:
Skip AMP test on WordPress.com for now.

jeherve commented:
This is on hold pending review on D27776-code.

jeherve commented:
The changes have now been committed to WordPress.com. This PR will bring Jetpack in sync for good, for all the files.

zinigor reviewed (approved):
Great to see this all synced with WordPress.com, let's get it in!
